### PR TITLE
[docs] Improve text, fix typos in Template Reference

### PR DESCRIPTION
**Pull request by straversi**
A couple typo fixes and phrase rewords.

### CODE REVIEW

File: docs/_guide/03-template-reference.md [hunk 1]
```diff
@@ -508,9 +508,9 @@ Takes a series of values, including Promises. Values are rendered in priority or
  lowest priority. If a value is a Promise, a lower-priority value will be rendered until it resolves.
 
 The priority of values can be used to create placeholder content for async
-data. For example, a Promise with pending content can be the first,
-highest-priority, argument, and a non-promise loading indicator template can
-be used as the second, lower-priority, argument. The loading indicator 
+data. For example, a Promise with pending content can be the first
+(highest-priority) argument, and a non-promise loading indicator template can
+be used as the second (lower-priority) argument. The loading indicator
```
[the remark this answers is not in the document]
Comment (keanulee) [reply docs/_guide/03-template-reference.md]: There appears to be a dangling backquote at the end of line 356 that should be removed.

[the remark this answers is not in the document]
Comment (straversi) [reply docs/_guide/03-template-reference.md]: Good eye. There's also a missing quote on 472 and an extraneous quote on 464. This PR seems like the place to fix all that.

[the remark this answers is not in the document]
Comment (straversi) [reply docs/_guide/03-template-reference.md]: Didn't seem to fix the highlighting though 🤔

**Comment by straversi** [comment]
A squash and merge would look nice in the git history 🙂

**Pull request merged**